### PR TITLE
Issue/969

### PR DESCRIPTION
**Pull request by valdar**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
Need backport to camel master

File: core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java [hunk 1]
```diff
@@ -177,31 +207,46 @@ private long remaining(long startPollEpochMilli, long maxPollDuration)  {
...
 
+            // take in account Cached camel streams
+            if (messageBodyValue instanceof StreamCache) {
+                StreamCache sc = (StreamCache) messageBodyValue;
+                // reset to be sure that the cache is ready to be used before sending it in the record (could be useful for SMTs)
+                sc.reset();
+                try {
+                    messageBodyValue = sc.copy(exchange);
+                } catch (IOException e) {
+                    e.printStackTrace();
```
Comment (lburgazzoli) [inline core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java:225]: log or rethrow

File: core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java [hunk 2]
```diff
@@ -210,6 +255,18 @@ private long remaining(long startPollEpochMilli, long maxPollDuration)  {
         return records.isEmpty() ? null : records;
     }
 
+    @Override
+    public void commitRecord(SourceRecord record, RecordMetadata metadata) throws InterruptedException {
```
Comment (lburgazzoli) [inline core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java:259]: who is throwing InterrupedException ? we need to be sure not to left an exchange uncompleted and not tracked any more in the array

File: core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java (hunk 1 above)
Comment (valdar) [reply core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java:225]: done in https://github.com/apache/camel-kafka-connector/pull/1093

File: core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java (hunk 2 above)
Comment (valdar) [reply core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java:259]: done in https://github.com/apache/camel-kafka-connector/pull/1093

**Comment by valdar** [comment]
> Need backport to camel master 

backporting here: https://github.com/apache/camel-kafka-connector/pull/1094

**Pull request merged**